jupyter_client/manager.py

```diff
@@ -337,15 +340,24 @@ def finish_shutdown(self, waittime=None, pollinterval=0.1):
                 self.log.debug("Kernel is taking too long to finish, killing")
                 self._kill_kernel()
 
-    def cleanup(self, connection_file=True):
+    def cleanup_resources(self, restart=False):
         """Clean up resources when the kernel is shut down"""
-        if connection_file:
+        if not restart:
             self.cleanup_connection_file()
 
         self.cleanup_ipc_files()
         self._close_control_socket()
         self.session.parent = None
 
+        if self._created_context and not restart:
+            self.context.destroy(linger=100)
+
+    def cleanup(self, connection_file=True):
+        """Clean up resources when the kernel is shut down"""
+        warnings.warn("Method cleanup(connection_file=True) is deprecated, use cleanup_resources(restart=False).",
+                      DeprecationWarning)
```

Comment:
As noted in this [previous comment](https://github.com/jupyter/jupyter_client/pull/548#issuecomment-636221159), I'm not seeing any warning messages produced in the server console/logs.  I think we should adjust the class such that it produces at least one message.  (Using `FutureWarning` does just that.)

Reply:
Done.